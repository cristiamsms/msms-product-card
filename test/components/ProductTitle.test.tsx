import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductCard';
import { ProductTitle } from '../../src/components/ProductTitle';
import { product1 } from '../data/product';




describe('ProductTitle', () => {
  test('debe de mostrar el componente correctamente con el titulo personalizado', () => {
    const wrapper =renderer.create(
        <ProductTitle title='test product' className='prueba stilos' style={{background:"white"}} />
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('debe mostrar el componente con el nombre del producto', () => {
    const wrapper =renderer.create(
        <ProductCard product={product1}>
            {
                ()=>(<ProductTitle/>)
            }
        </ProductCard>
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
  
});



 