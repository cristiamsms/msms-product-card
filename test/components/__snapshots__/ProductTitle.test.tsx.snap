// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el componente correctamente con el titulo personalizado 1`] = `
<span
  className="productDescription prueba stilos"
  style={
    Object {
      "background": "white",
    }
  }
>
  test product
</span>
`;

exports[`ProductTitle debe mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - no imagen
  </span>
</div>
`;
